CMakeLists.txt

```diff
@@ -712,10 +712,10 @@ create_source_groups(${TEST_FILES})
 
 target_link_libraries(
 	${TEST_PROJECT_NAME}
-	${LIB_PROJECT_NAME}
-	${LIB_GUI_PROJECT_NAME}
 	$<$<BOOL:${BUILD_CXX_LANGUAGE_PACKAGE}>:${LIB_CXX_PROJECT_NAME}>
 	$<$<BOOL:${BUILD_JAVA_LANGUAGE_PACKAGE}>:${LIB_JAVA_PROJECT_NAME}>
+	${LIB_PROJECT_NAME}
+	${LIB_GUI_PROJECT_NAME}
```

Comment:
`${LIB_GUI_PROJECT_NAME}` should still stay on top because it depends on all others. That might not be important for the test target right now, but could become important later.

Reply:
Ok, just read your explanation above why `lib_gui` is below `lib`. We would like to get rid of the mentioned dependency. I will merge it now.

Reply:
See my previous comment `${LIB_PROJECT_NAME}` is using functions from `${LIB_GUI_PROJECT_NAME}` like `utility::killRunningProcesses()` or `utility::getIdealThreadCount()` defined in `src/lib_gui/utility/utilityApp.cpp`.

This means that `${LIB_GUI_PROJECT_NAME}` has to be provided after `${LIB_PROJECT_NAME}`. 

The real issue here is you have circular dependencies:

- `${LIB_PROJECT_NAME}` is using `${LIB_GUI_PROJECT_NAME}`
- `${LIB_GUI_PROJECT_NAME}` is using `${LIB_CXX_PROJECT_NAME}`
- `${LIB_CXX_PROJECT_NAME}` is using `${LIB_PROJECT_NAME}`

If you want to have `${LIB_GUI_PROJECT_NAME}` on top, you must refactor your code to untangle the circular dependencies.